DataFormats/HcalDigi/test/test_hcal_digi.cu

```diff
@@ -0,0 +1,183 @@
+        samples + TDF::HEADER_WORDS + TDF::FLAG_WORDS) * sizeof(uint16_t), 
+            cudaMemcpyHostToDevice);
+    kernel_test_hcal_qie1011_digis<TDF><<<1, size>>>(d_data, d_out, samples);
+    cudaDeviceSynchronize();
+    auto code = cudaGetLastError();
+    if (code != cudaSuccess)
+        std::cout << cudaGetErrorString(code);
+    cudaMemcpy(&h_out, d_out, size * sizeof(uint32_t), cudaMemcpyDeviceToHost);
+
+    // comparison
+    for (auto i=0; i<size; i++) {
+        std::cout << h_out[i] << " == " << h_test_out[i] << std::endl;
+        assert(h_out[i] == h_test_out[i]);
+    }
+}
+
+
+
+template<typename TDF>
+void test_hcal_qie8_digis() {
+    constexpr int n = 10;
+    edm::SortedCollection<TDF> coll{n};
+    TDF *d_dfs;
+    uint32_t *d_out;
+    uint32_t h_out[n], h_test_out[n];
+    for (auto i=0; i<n; i++) {
+        TDF &df = coll[i];
+        df.setSize(10);
+        h_test_out[i] = 0;
+        uint32_t test = 0;
+        for (auto j=0; j<10; j++) {
+            df.setSample(j, HcalQIESample(100));
+            h_test_out[i] += df.sample(j).raw();
+            test += df.sample(j).raw();
+        }
+    }
+
+    cudaMalloc((void**)&d_dfs, n * sizeof(TDF));
+    cudaMalloc((void**)&d_out, n * sizeof(uint32_t));
+    cudaMemcpy(d_dfs, &(*coll.begin()), n * sizeof(TDF), 
+        cudaMemcpyHostToDevice);
+    kernel_test_hcal_qie8_digis<<<1, n>>>(d_dfs, d_out);
+    cudaMemcpy(&h_out, d_out, n * sizeof(uint32_t), cudaMemcpyDeviceToHost);
+
+    std::cout << "collection size = " << coll.size() << std::endl;
+
+    // comparison
+    for (auto i=0; i<n; i++) {
+        std::cout << h_out[i] << " == " << h_test_out[i] << std::endl;
+        assert(h_out[i] == h_test_out[i]);
+    }
+}
+
+void test_hcal_qie8_hbhedf() {
+    HBHEDataFrame h_df, h_test_df;
+    HBHEDataFrame *d_df;
+
+    h_test_df.setSize(10);
+    for (auto i=0; i<10; i++)
+        h_test_df.setSample(i, HcalQIESample(100));
+    h_test_df.setReadoutIds(HcalElectronicsId(100));
+
+    cudaMalloc((void**)&d_df, sizeof(HBHEDataFrame));
+    cudaMemcpy(d_df, &h_df, sizeof(HBHEDataFrame), cudaMemcpyHostToDevice);
+    kernel_test_hcal_qie8_hbhedf<<<1,1>>>(d_df);
+    cudaMemcpy(&h_df, d_df, sizeof(HBHEDataFrame), cudaMemcpyDeviceToHost);
+
+    assert(h_df.size() == h_test_df.size());
+    assert(h_df.elecId() == h_test_df.elecId());
+    for (auto i=0; i<10; i++)
+        assert(h_df[i].raw() == h_test_df[i].raw());
+}
+
+int main(int argc, char** argv) {
+    int nDevices;
+    cudaGetDeviceCount(&nDevices);
+    std::cout << "nDevices = " << nDevices << std::endl;
+
+    if (nDevices > 0) {
+        // qie8
+        test_hcal_qiesample();
+        test_hcal_qie8_hbhedf();
+        test_hcal_qie8_digis<HBHEDataFrame>();
+        test_hcal_qie8_digis<HFDataFrame>();
+        test_hcal_qie8_digis<HODataFrame>();
+
+        // qie1011
+        test_hcal_qie1011_digis<QIE10DataFrame>();
+        test_hcal_qie1011_digis<QIE11DataFrame>();
+    }
+}
```

Comment:
Could you add `return 0` since the `main` is declared to return an `int`?

Reply:
done